updates to RELEASE.txt and  a comment in version.cs (#1733)
dafny-lang/dafny: RELEASE.txt

1) In an up-to-date repository, with the master branch checked out
   and up-to-date (e.g., git pull upstream)
   checkout a new branch from master (e.g., git checkout -b branchname master)
   cd to the repo root (i.e., dafny, which contains Source, Binaries, etc.)

2) Select a version number (descriptor) $VER (e.g., "3.0.0" or "3.0.0-alpha")

>    and select an internal version number (e.g., "3.0.0.30203")
>    guideline for internal version number:Use the no. of years since 2018 and release date
>    e.g.,Release on January 12th 2022 = 2018 + 4 years and date in "mmdd" - x.y.z.40112

Reviewer: ```suggestion
   and select an internal version number (e.g., "3.0.0.30203").
   Obtain it by concatenating the digits in the release date,
   e.g.,Release on January 12th 2022 = 2018 + 4 years and date in "mmdd" - x.y.z.40112
```

Reviewer: Sorry, I changed my mind after. Feel free to adopt what makes sense to you.

1) In an up-to-date repository, with the master branch checked out
   and up-to-date (e.g., git pull upstream)
   checkout a new branch from master (e.g., git checkout -b branchname master)
   cd to the repo root (i.e., dafny, which contains Source, Binaries, etc.)

2) Select a version number (descriptor) $VER (e.g., "3.0.0" or "3.0.0-alpha")

>    and select an internal version number (e.g., "3.0.0.30203")
>    The last section is in format "ymmdd" format
>    where y is years since 2018 and then release date in "mmdd"
>    (e.g., Release on January 12th 2022  2018 + 4 years and date in "mmdd" - x.y.z.40112)

Reviewer: ```suggestion
   and select an internal version number (e.g., "3.0.0.30203").
   The last section is in "ymmdd" format, where "y" is the number of years
   since 2018 and "mmdd" the month and day portions of the release date (e.g.,
   release on January 12th, 2022 -> x.y.z.40112).
```

1) In an up-to-date repository, with the master branch checked out
   and up-to-date (e.g., git pull upstream)

>    checkout a nre branch from master (e.g., git checkout -b branchname master)

Reviewer: ```suggestion
   checkout a new branch from master (e.g., git checkout -b branchname master)
```

5) In the Github releases, edit the new release to add in the release notes.

6) Push the Publish button.

>    if doing a pre-release stop here, as steps further are to define the release as
>    latest version to be installed which is not needed.

Reviewer: ```suggestion
   If preparing a pre-release, stop here, as the following steps declare the release
   as the latest version, which is not the intention.
```
